File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/repository/configuration/IUserInfoRepository.java
package ru.usedesk.chat_sdk.internal.data.repository.configuration;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import ru.usedesk.chat_sdk.external.entity.UsedeskChatConfiguration;
import ru.usedesk.common_sdk.external.entity.exceptions.UsedeskDataNotFoundException;


public interface IUserInfoRepository {

    @NonNull
    String getToken() throws UsedeskDataNotFoundException;

    void setToken(@Nullable String token);

    @NonNull
    UsedeskChatConfiguration getConfiguration() throws UsedeskDataNotFoundException;

    void setConfiguration(@Nullable UsedeskChatConfiguration configuration);
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/repository/configuration/IUserInfoRepository.kt
package ru.usedesk.chat_sdk.internal.data.repository.configuration

import ru.usedesk.chat_sdk.external.entity.UsedeskChatConfiguration
import ru.usedesk.common_sdk.external.entity.exceptions.UsedeskDataNotFoundException

interface IUserInfoRepository {
    @Throws(UsedeskDataNotFoundException::class)
    fun getToken(): String

    fun setToken(token: String?)

    @Throws(UsedeskDataNotFoundException::class)
    fun getConfiguration(): UsedeskChatConfiguration

    fun setConfiguration(configuration: UsedeskChatConfiguration?)
}